### PR TITLE
Deploy/deploy scripts

### PR DESCRIPTION
**Pull request by j-vp**

### CODE REVIEW

File: script/Deploy.s.sol [hunk 1]
```diff
@@ -395,25 +397,22 @@ contract ArcadiaVaultDeployer is Test {
...
-
         pool_weth =
-        new LendingPool(ERC20(address(weth)), 0x12e463251Bc79677FD980aA6c301d5Fb85101cCb, address(factory), address(0));
-        srTranche_weth = new Tranche(address(pool_weth), "Senior", "SR");
-        jrTranche_weth = new Tranche(address(pool_weth), "Junior", "JR");
+            new LendingPool(ERC20(address(weth)), DeployAddresses.treasury, address(factory), address(liquidator));
+        srTranche_weth = new Tranche(address(pool_weth), "Senior", "s");
+        jrTranche_weth = new Tranche(address(pool_weth), "Junior", "j");
 
         pool_weth.setVaultVersion(1, true);
```
Comment (Thomas-Smets) [inline script/Deploy.s.sol:408]: Also set the fixedLiquidationCost

File: script/Deploy.s.sol [hunk 2]
```diff
@@ -395,25 +397,22 @@ contract ArcadiaVaultDeployer is Test {
...
         pool_weth.setOriginationFee(10);
+        pool_weth.setMaxInitiatorFee(66 * 10 ** 18);
         pool_weth.addTranche(address(srTranche_weth), 50, 0);
         pool_weth.addTranche(address(jrTranche_weth), 40, 20);
         pool_weth.setTreasuryInterestWeight(10);
         pool_weth.setTreasuryLiquidationWeight(80);
```
Comment (Thomas-Smets) [inline script/Deploy.s.sol:414]: Only give 20% of liquidations to junior tranche? Would bump it in the beginning

File: src/Liquidator.sol [hunk 3]
```diff
@@ -96,8 +96,8 @@ contract Liquidator is Owned {
         factory = factory_;
         initiatorRewardWeight = 1;
         penaltyWeight = 5;
-        startPriceMultiplier = 110;
-        minPriceMultiplier = 50;
+        startPriceMultiplier = 150;
+        minPriceMultiplier = 60;
         cutoffTime = 14_400; //4 hours
         base = 1e18;
```
Comment (Thomas-Smets) [inline src/Liquidator.sol:102]: Can't set this to 1e18! should be smaller than 1

File: script/Deploy_lending_mainnet.s.sol [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+        vm.startBroadcast(deployerPrivateKey);
+
+        factory = new Factory();
+        liquidator = new Liquidator(address(factory));
+        liquidator.setAuctionCurveParameters(3600, 14_400);
+
+        pool_weth =
+            new LendingPool(ERC20(address(weth)), DeployAddresses.treasury, address(factory), address(liquidator));
+        srTranche_weth = new Tranche(address(pool_weth), "Senior", "s");
```
Comment (Thomas-Smets) [inline script/Deploy_lending_mainnet.s.sol:51]:
```suggestion
        srTranche_weth = new Tranche(address(pool_weth), "Senior", "sr");
```

File: script/Deploy_lending_mainnet.s.sol [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+        jrTranche_weth = new Tranche(address(pool_weth), "Junior", "j");
```
Comment (Thomas-Smets) [inline script/Deploy_lending_mainnet.s.sol:52]:
```suggestion
        jrTranche_weth = new Tranche(address(pool_weth), "Junior", "jr");
```

File: script/Deploy_lending_mainnet.s.sol [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+        pool_weth.setOriginationFee(10);
+        pool_weth.setMaxInitiatorFee(66 * 10 ** 18);
```
Comment (Thomas-Smets) [inline script/Deploy_lending_mainnet.s.sol:55]: Too big

File: script/Deploy_lending_mainnet.s.sol [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+            })
+        );
+
+        pool_usdc =
+            new LendingPool(ERC20(address(usdc)), DeployAddresses.treasury, address(factory), address(liquidator));
+        srTranche_usdc = new Tranche(address(pool_usdc), "Senior", "s");
+        jrTranche_usdc = new Tranche(address(pool_usdc), "Junior", "j");
+
+        pool_usdc.setOriginationFee(10);
+        pool_usdc.setMaxInitiatorFee(100_000 * 10 ** 6);
```
Comment (Thomas-Smets) [inline script/Deploy_lending_mainnet.s.sol:76]: Too big

File: script/Deploy_lending_optimism.s.sol [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+        srTranche_weth = new Tranche(address(pool_weth), "Senior", "s");
+        jrTranche_weth = new Tranche(address(pool_weth), "Junior", "j");
+
+        pool_weth.setOriginationFee(10);
+        pool_weth.setMaxInitiatorFee(66 * 10 ** 18);
+        pool_weth.addTranche(address(srTranche_weth), 50, 0);
+        pool_weth.addTranche(address(jrTranche_weth), 40, 20);
+        pool_weth.setTreasuryInterestWeight(10);
+        pool_weth.setTreasuryLiquidationWeight(80);
+        pool_weth.setSupplyCap(1);
```
Comment (Thomas-Smets) [inline script/Deploy_lending_optimism.s.sol:60]: setBorrowCap as well?

File: script/Deploy_lending_optimism.s.sol [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+        liquidator = new Liquidator(address(factory));
+        liquidator.setAuctionCurveParameters(3600, 14_400);
+
+        pool_weth =
+            new LendingPool(ERC20(address(weth)), DeployAddresses.treasury, address(factory), address(liquidator));
+        srTranche_weth = new Tranche(address(pool_weth), "Senior", "s");
+        jrTranche_weth = new Tranche(address(pool_weth), "Junior", "j");
+
+        pool_weth.setOriginationFee(10);
+        pool_weth.setMaxInitiatorFee(66 * 10 ** 18);
```
Comment (Thomas-Smets) [inline script/Deploy_lending_optimism.s.sol:55]: Should be much much lower on optimism?
set, fixedLiquidationCost also, lower as on mainnet

File: script/Deploy_lending_optimism.s.sol [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+    }
+
+    function run() public {
+        uint256 deployerPrivateKey = vm.envUint("PRIVATE_KEY_DEPLOYER");
+
+        vm.startBroadcast(deployerPrivateKey);
+
+        factory = new Factory();
+        liquidator = new Liquidator(address(factory));
+        liquidator.setAuctionCurveParameters(3600, 14_400);
```
Comment (Thomas-Smets) [inline script/Deploy_lending_optimism.s.sol:47]: Set in constructor?

File: script/Deploy_lending_mainnet.s.sol [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+    }
+
+    function run() public {
+        uint256 deployerPrivateKey = vm.envUint("PRIVATE_KEY_DEPLOYER");
+
+        vm.startBroadcast(deployerPrivateKey);
+
+        factory = new Factory();
+        liquidator = new Liquidator(address(factory));
+        liquidator.setAuctionCurveParameters(3600, 14_400);
```
Comment (Thomas-Smets) [inline script/Deploy_lending_mainnet.s.sol:47]: Set in constructor

File: script/Deploy.s.sol (hunk 2 above)
Comment (j-vp) [reply script/Deploy.s.sol:414]: bumped to 50%

File: script/Deploy_lending_mainnet.s.sol (hunk 7 above)
Comment (j-vp) [reply script/Deploy_lending_mainnet.s.sol:76]: halved it to 50k

File: script/Deploy_lending_optimism.s.sol (hunk 9 above)
Comment (j-vp) [reply script/Deploy_lending_optimism.s.sol]: 3 eth / 5k usdc on optimism, 33 eth / 50k on mainnet

File: src/Liquidator.sol (hunk 3 above)
Comment (j-vp) [reply src/Liquidator.sol:102]: set to 999807477651317446 = 3600s halflife, 14_400 cutoff

File: script/Deploy_lending_mainnet.s.sol (hunk 6 above)
Comment (j-vp) [reply script/Deploy_lending_mainnet.s.sol]: new params: mainnet: 33 eth & 50k usdc, optimism: 3eth & 5k usdc

**Review by Thomas-Smets — APPROVED** [review]
gtg

**Pull request merged**